qemu/tests/migration_multi_host_with_kdump.py

```diff
@@ -71,6 +72,70 @@ def __init__(self, test, params, env):
 
         @error.context_aware
+        def start_worker_guest_kdump(self, mig_data, login_timeout,
+                                     crash_kernel_prob_cmd,
+                                     kernel_param_cmd,
+                                     kdump_enable_cmd,
+                                     nvcpu, crash_cmd):
+
+            """
+            force the Linux kernel to crash before migration
+
+            :param mig_data: Data for migration
+            :param login_timeout: timeout of login
+            :param crash_kernel_prob_cmd: cmd for check kdump loaded
+            :param kernel_param_cmd: the param add into kernel line for kdump
+            :param kdump_enable_cmd: enable kdump command
+            :param nvcpu: which is used to trigger a crash
+            :param crash_cmd: which is triggered crash command
+            """
+
+            vm = mig_data.vms[0]
+            kdump.preprocess_kdump(vm, login_timeout)
+            kdump.kdump_enable(vm, vm.name, crash_kernel_prob_cmd,
+                               kernel_param_cmd, kdump_enable_cmd,
+                               login_timeout)
+            error.context("Kdump Testing, force the Linux kernel to crash",
+                          logging.info)
+            kdump.crash_test(vm, nvcpu, crash_cmd, login_timeout)
+
+        @error.context_aware
+        def check_worker_kdump(self, mig_data, vmcore_chk_cmd, vmcore_incomplete):
+
+            """
+            check weather generate vmcore file after migration
+
+            :param mig_data: Data for migration.
+            :param vmcore_chk_cmd: cmd for check vmcore file
+            :param vmcore_incomplete: the name of vmcore file when error
+            """
+
+            if not self.is_src:
+                for vm in mig_data.vms:
+                    if vm.is_paused():
+                        vm.resume()
+                    if not utils_test.qemu.guest_active(vm):
+                        raise error.TestFail("Guest not active "
+                                             "after migration")
+                    logging.info("Logging into migrated guest after "
+                                 "migration")
+                    session = vm.wait_for_login(timeout=self.login_timeout)
+                    error.context("Checking vmcore file in guest",
+                                  logging.info)
+                    if session is not None:
+                        logging.info("kdump completed, no need ping-pong"
+                                     " migration")
+                        self.stop_migrate = True
+                    output = session.cmd_output(vmcore_chk_cmd)
+                    kdump.postprocess_kdump(vm, self.login_timeout)
+                    if not output:
+                        raise error.TestFail("Could not found vmcore file")
+                    elif vmcore_incomplete in output.split("\n"):
+                        raise error.TestError("Kdump is failed")
```

Comment:
TestError is for those env preparation failed scenarios.
So I think should use TestFail here.